File: Maths/ProdOfTwoPrimes.py
# Product of Two Prime Numbers
# https://edabit.com/challenge/LuC3AwFKJg2eEA4Kz

import math

def f(n:int)->bool:
	s = []
	x = 2
	while(x<=(int)(math.sqrt(n))):
		while(n%x==0):
			s.append(x)
			n/=x
		x+=1
	if(n>1):
		s.append(n)
	if(len(s)!=2):
		return False
	return True


if __name__=='__main__':
	n = 5767
	print(f(n),end="\n") # True
	
	

File: Maths/ProdOfTwoPrimes.cxx
// Product of Two Prime Numbers
// https://edabit.com/challenge/LuC3AwFKJg2eEA4Kz

#include <bits/stdc++.h>
using namespace std;

bool f(int n){
	vector<int> s;
	auto x = 2;
	while(x <= sqrt(n)){
		while(n%x==0){
			s.emplace_back(x);
			n/=x;
		}
		x++;
	}
	if(n > 1)
		s.emplace_back(n);
	if(s.size()!=2)
		return false;
	return true;
}

int main(){
	int n = 5767;
	printf("%d\n",f(n)); // True
	return 0;
}	
